test/system/groups_test.rb

```diff
@@ -122,6 +122,37 @@ def setup
     assert_selector 'tr', count: @groups_count - 1
   end
 
+  test 'can transfer a group' do
+    group = groups(:group_one)
+    visit group_url(group)
+
+    click_link I18n.t(:'groups.sidebar.settings')
+    assert_selector 'h3', text: I18n.t(:'groups.edit.advanced.transfer.title')
+
+    within %(form[action="/group-1/transfer"]) do
+      find('#new_namespace_id').find(:xpath, 'option[14]').select_option
+      click_on I18n.t(:'groups.edit.advanced.transfer.submit')
+    end
+
+    within('#turbo-confirm') do
+      assert_text I18n.t(:'components.confirmation.title')
+      find('input[type=text]').fill_in with: group.path
+      click_on I18n.t(:'components.confirmation.confirm')
+    end
+
+    assert_text I18n.t(:'groups.transfer.success')
+  end
+
+  test 'user with maintainer access should not be able to see the transfer group section' do
+    login_as users(:joan_doe)
+
+    visit group_url(groups(:group_one))
+
+    click_link I18n.t(:'groups.sidebar.settings')
+
+    assert_selector 'h3', text: I18n.t(:'groups.edit.advanced.transfer.title'), count: 0
+  end
+
```

Comment:
Can I also get you to add on a test here which displays and asserts an error message when transferring?

Reply:
I am all for improving test coverage. Please see df20d0d899cf1f04484a7c99fa7962d2e074fc24. But projects/transfer_test.rb only require 1 test to be merged.

Reply:
projects/transfer_test.rb was updated in !139 to include a test for the same error condition. This was missed on initial review and @deepsidhu85 found it in a followup review.